### ISSUE TITLE
Using JSON base values as API request body / response body?

### ISSUE THREAD

**Issue by cheppsn** [stoicflame/enunciate#404]
We use Spring MVC with API paths such as `PUT /api/users/1/email` with a plain JSON String value as body of the request, e.g. `"john@doe.com"`.

To map the plain JSON String value to a controller method argument, we use Spring's `TypedStringValue` class, e.g.

    @RequestMapping(method = RequestMethod.PUT, path = "/users/{userId}/email")
	@ResponseStatus(code = HttpStatus.NO_CONTENT)
	public void putEmail(
			@PathVariable("userId") Integer userId,
			@RequestBody TypedStringValue newEmail) throws NotFoundException
	{
		this.users.setEmail(userId, newEmail.getValue())
				.orElseThrow(NotFoundException::new);
	}

Enunciate is configured to use the following modules:
- spring-web
- jackson

If we generate Docs with Enunciate for such a call, the `TypedStringValue` argument is interpreted as POJO and we receive something like

    PUT /api/users/{userId}/email
    
    Request Parameters
    
    name              type   description   constraints
    ----------------------------------------------------
    userId            path                 int 
    source            query    
    specifiedTypeName query    
    targetType        query    
    targetTypeName    query    
    value             query    
    

    Request Body
    
    media type        data type
    ----------------------------------------------------
    application/json  TypedStringValue (JSON) 
    
    
    JSON TypedStringValue Data Type
    
    Properties
    
    name               data type    description
    ----------------------------------------------------
    specifiedTypeName  string   
    value              string   
    source             object   
    targetType         object   
    targetTypeName     string   
    dynamic            boolean  

However, what I would like to receive is:

    PUT /api/users/{userId}/email
    
    Request Parameters
    
    name              type   description   constraints
    ----------------------------------------------------
    userId            path                 int 
    

    Request Body
    
    media type        data type
    ----------------------------------------------------
    application/json  string (JSON) 

There are two issues with this:

1. Why is the `@RequestBody` argument resulting in `query` Request Parameter(s)? This seems to be a bug.
1. Is there a way to override/extend the generation of the `Request Body` section (to replace `TypedStringValue`with `string`) and to suppres generation of JSON type altogether for `TypedStringValue`?

I'm aware that there is a debate on whether it is OK to send plain JSON values (instead of objects / lists) in REST calls, but to us this makes sense for PUT, if an actual property of an object is replaced. 

So I wonder whether there is a way to adapt Enunciate to match our expectations?

**Comment by stoicflame** [comment]
The fact that the properties of `TypedStringValue` showed up as query parameters was a bug. Fixed at 8f3b755.

You can give Enunciate a hint about `TypedStringValue` by annotating the parameter with `@com.webcohesion.enunciate.metadata.rs.TypeHint`:

```java
@RequestMapping(method = RequestMethod.PUT, path = "/users/{userId}/email")
@ResponseStatus(code = HttpStatus.NO_CONTENT)
public void putEmail(
        @PathVariable("userId") Integer userId,
        @RequestBody @TypeHint(String.class) TypedStringValue newEmail) throws NotFoundException
{
    this.users.setEmail(userId, newEmail.getValue())
            .orElseThrow(NotFoundException::new);
}
```

**Comment by stoicflame** [comment]
[Enunciate 2.4 has been released](https://github.com/stoicflame/enunciate/releases/tag/v2.4.0).
